From 2a74825cf54677c54a9231abbb3856a7027edf36 Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Mon, 20 Sep 2010 14:22:30 +0800
Subject: [PATCH 080/635] ENGR00131744 The camera may crash when switch from
 thumbnai mode to preview mode

The cause is that setposition may set the negative value for x, y.
It will crash the IPU. So make a walkaround here.

   Signed-off-by: Yaoming Qin <b10945@freescale.com>
---
 liboverlay/overlay.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/liboverlay/overlay.cpp b/liboverlay/overlay.cpp
index fbce82a..bf0f256 100755
--- a/liboverlay/overlay.cpp
+++ b/liboverlay/overlay.cpp
@@ -1052,6 +1052,12 @@ static int overlay_setPosition(struct overlay_control_device_t *dev,
     /* set this overlay's position (talk to the h/w) */
     overlay_object *obj = static_cast<overlay_object *>(overlay);
 
+    if (x <0 || y <0 )
+    {
+        OVERLAY_LOG_ERR ("!!!!Overlay pos set: x %d,y %d,w %d,h %d",x,y,w,h);
+        return 0;
+    }
+
     //fetch the overlay obj lock
     pthread_mutex_lock(&obj->mDataShared->obj_lock);
     if((x!= obj->outX)||
-- 
1.8.0

